### PR TITLE
Filter on click

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## Summary
* Adds a LiveSearch module, based heavily on the GOVUK equivalent (https://github.com/alphagov/finder-frontend/blob/67760d95a6b97980db02c72170de647948160926/app/assets/javascripts/live_search.js) but with non-trivial changes for our setup.
* Tweak OptionSelects to only display their `<x> selected` label when collapsed.
* Add missing dependency on `jasmine-core` (bugfix).
* Vendor jasmine-jquery and mock-ajax (for testing).
* Bump version to 22.5.0

## Ticket
https://trello.com/c/SGxQeBdA/502-filter-on-click

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "del": "^2.2.2",
     "govuk_frontend_toolkit": "5.0.3",
     "gulp": "3.8.7",
-    "gulp-jasmine-phantom": "2.0.1"
+    "gulp-jasmine-phantom": "2.0.1",
+    "jasmine-core": "2.6.4"
```
Comment (CrystalPea) [inline package.json]: bumping versions looks right 👍

File: spec/javascripts/manifest.js [hunk 2]
```diff
@@ -3,19 +3,24 @@ var manifest = {
   support : [
     'vendor/jquery.js',
     'vendor/hogan.js',
+    'vendor/jasmine-2.0.1/mock-ajax.js',
+    'vendor/jasmine-2.0.1/jasmine-jquery.js',
```
Comment (CrystalPea) [inline spec/javascripts/manifest.js]: this looks alright, too

File: spec/javascripts/manifest.js [hunk 3]
```diff
@@ -3,19 +3,24 @@ var manifest = {
...
     '../../../node_modules/govuk_frontend_toolkit/javascripts/vendor/polyfills/bind.js',
     '../../../toolkit/javascripts/list-entry.js',
     '../../../toolkit/javascripts/module-loader.js',
     '../../../toolkit/javascripts/option-select.js',
     '../../../toolkit/javascripts/show-hide-content.js',
     '../../../toolkit/javascripts/validation.js',
-    '../../../toolkit/javascripts/word-counter.js'
+    '../../../toolkit/javascripts/word-counter.js',
+    '../../../toolkit/javascripts/support.js',
+    '../../../toolkit/javascripts/live-search.js'
```
Comment (CrystalPea) [inline spec/javascripts/manifest.js]: 👍

File: spec/javascripts/manifest.js [hunk 4]
```diff
@@ -3,19 +3,24 @@ var manifest = {
...
   ],
   test : [
     '../unit/ListEntrySpec.js',
     '../unit/WordCountSpec.js',
     '../unit/show-hide-content.spec.js',
     '../unit/validation.spec.js',
+    '../unit/live-search.js',
```
Comment (CrystalPea) [inline spec/javascripts/manifest.js]: 👍

File: toolkit/javascripts/option-select.js [hunk 5]
```diff
@@ -80,10 +80,13 @@
   };
 
   OptionSelect.prototype.checkedString = function checkedString(){
-    var count = this.$options.filter(":checked").size();
     var checkedString = "";
-    if (count > 0){
-      checkedString = count+" selected";
+
+    if (this.isClosed()) {
```
Comment (CrystalPea) [inline toolkit/javascripts/option-select.js]: Does this line of code tell to perform filtering automatically if filtering dialogue is closed?

File: toolkit/javascripts/support.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+if(typeof window.GOVUK === 'undefined'){ window.GOVUK = {}; }
+if(typeof window.GOVUK.GDM === 'undefined'){ window.GOVUK.GDM = {}; }
+if(typeof window.GOVUK.GDM.support === 'undefined'){ window.GOVUK.GDM.support = {}; }
+
+window.GOVUK.GDM.support.history = function() {
+  return window.history && window.history.pushState && window.history.replaceState;
+};
```
Comment (CrystalPea) [inline toolkit/javascripts/support.js]: This looks alright

File: package.json (hunk 1 above)
Comment (samuelhwilliams) [inline package.json]: No, it doesn't. If it's closed, the added line (102) in toggleOptionSelect triggers the update when it's actually closed.

File: toolkit/javascripts/option-select.js (hunk 5 above)
Comment (samuelhwilliams) [inline toolkit/javascripts/option-select.js]: No, it doesn't. If it's closed, the added line (102) in toggleOptionSelect triggers the update when it's actually closed.
